velox/type/Type.h

```diff
@@ -1389,8 +1389,6 @@ std::shared_ptr<const OpaqueType> OPAQUE() {
   [&]() {                                                                      \
     if ((typeKind) == ::facebook::velox::TypeKind::UNKNOWN) {                  \
       return TEMPLATE_FUNC<::facebook::velox::TypeKind::UNKNOWN>(__VA_ARGS__); \
-    } else if ((typeKind) == ::facebook::velox::TypeKind::OPAQUE) {            \
-      return TEMPLATE_FUNC<::facebook::velox::TypeKind::OPAQUE>(__VA_ARGS__);  \
```

Comment:
It would be better to keep `VELOX_DYNAMIC_TYPE_DISPATCH_ALL` unchanged. Maybe, we can define a new macro, e.g., `VELOX_DYNAMIC_TYPE_DISPATCH_ALL_EXCEPT_OPAQUE`.

Reply:
Yeah， Maybe it's a good workaround. But it seems to be a bug in `VELOX_DYNAMIC_TYPE_DISPATCH_ALL` which can't initialize OPAQUE(void * can't be referenced)

Reply:
`VELOX_DYNAMIC_TYPE_DISPATCH_ALL` seems applied to some specific functions, we can't change it. modify `VELOX_DYNAMIC_TYPE_DISPATCH` instead.